src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -458,20 +448,20 @@ object DepositHandler {
 
 
   // TODO: RETRIEVE VIA AN INTERFACE
-  private def getReferredBagChecksums(url: String)(implicit bs: BagStoreBase): Try[Seq[(String, String)]] =
+  private def getReferredBagChecksums(url: String)(implicit bs: Option[BagStoreSettings]): Try[Seq[(String, String)]] =
     getBag(getReferredBagDir(url)).map(bag => {
       bag.getPayloadManifests
         .asScala
         .flatMap(_.asScala)
     })
 
-  private def getReferredBagDir(url: String)(implicit bs: BagStoreBase): File = {
+  private def getReferredBagDir(url: String)(implicit bs: Option[BagStoreSettings]): File = {
     //  http://deasy.dans.knaw.nl/aips/31aef203-55ed-4b1f-81f6-b9f67f324c87.2/data/x -> 31/aef20355ed4b1f81f6b9f67f324c87/2
-    val Array(uuid, version) = url.stripPrefix(bs.baseUrl)
+    val Array(uuid, version) = url.stripPrefix(bs.get.baseUrl)
       .split("/data").head.replaceAll("-", "")
       .split("\\.")
     val (topDir, uuidDir) = uuid.splitAt(3)
 
-    getFile(bs.baseDir, topDir, uuidDir, version)
+    getFile(bs.get.baseDir, topDir, uuidDir, version)
```

Comment:
and here

Maybe you can give the `BagStoreBase` here as an implicit, rather than its `Option`. (**btw, it is a good thing to always try to avoid the `Option`, `Try`, etc. as an input parameter**). You can get it from somewhere, I guess